src/test/java/com/google/sps/ContextServletTest.java

```diff
@@ -0,0 +1,105 @@
+import javax.servlet.http.HttpServletResponse;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** This is a JUnit test for sentiment mockedAnalysis */
+@RunWith(JUnit4.class)
+public class ContextServletTest {
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+  private static final String URL_PARAMETER = "url";
+  private static final String TEST_URL = "WkZ5e94QnWk";
+  private static final String TEST_TITLE = "Test Title";
+  private static final String TEST_CHANNEL = "Test Channel";
+  private static final String REQUEST_INFO = "snippet,statistics";
+  private static final int NUM_LIKES_VAL = 10;
+  private static final int NUM_DISLIKES_VAL = 1;
+
+  @Test
+  public void testYoutubeAPIConstructed() throws IOException, GeneralSecurityException {
+    // Test the constructeAndExecuteYoutubeReuest method for returning a youtube request
+    YouTube mockedYoutubeService = mock(YouTube.class);
+    YouTube.Videos mockedVidoes = mock(YouTube.Videos.class);
+    YouTube.Videos.List mockedVideoList = mock(YouTube.Videos.List.class, RETURNS_DEEP_STUBS);
+    ContextServlet contestServlet = new ContextServlet(mockedYoutubeService);
+    when(mockedVidoes.list(REQUEST_INFO)).thenReturn(mockedVideoList);
+    when(mockedYoutubeService.videos()).thenReturn(mockedVidoes);
+    when(mockedVideoList.execute()).thenReturn(new VideoListResponse());
+    VideoListResponse response = contestServlet.constructAndExecuteYoutubeRequest(TEST_URL);
+    Assert.assertEquals(response.getClass(), VideoListResponse.class);
+    Assert.assertEquals(response.getItems().size(), 0);
+  }
+
+  @Test
+  public void testContextGet_generateExpectedResponse()
+      throws IOException, GeneralSecurityException, ServletException {
+    // Test the doGet() method to retrieve youtube vidoe context with mocked service
+    ContextServlet contextServlet = spy(ContextServlet.class);
+    HttpServletRequest mockedRequest = mock(HttpServletRequest.class);
+    HttpServletResponse mockedResponse = mock(HttpServletResponse.class);
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    VideoSnippet mockedVideoSnippet =
+        new VideoSnippet()
+            .setPublishedAt(new DateTime(new Date()))
+            .setTitle(TEST_TITLE)
+            .setChannelTitle(TEST_CHANNEL);
+    VideoStatistics mockedVidoeStatistics =
+        new VideoStatistics().setLikeCount(BigInteger.TEN).setDislikeCount(BigInteger.ONE);
+    Video mockedVideo =
+        new Video().setSnippet(mockedVideoSnippet).setStatistics(mockedVidoeStatistics);
+    VideoListResponse mockedVideoList =
+        new VideoListResponse().setItems(Collections.singletonList(mockedVideo));
+    when(mockedRequest.getParameter(URL_PARAMETER)).thenReturn(TEST_URL);
+    when(contextServlet.constructAndExecuteYoutubeRequest(TEST_URL)).thenReturn(mockedVideoList);
```

Comment:
Question: Why are we mocking constructAndExecuteYoutubeRequest? I thought we wanted to test this method.

Reply:
Hi Cis, the reason for me to mock this to test the entire doGet() method.  Rick and I discussed on this after the comments: https://github.com/googleinterns/step31-2020/pull/43#discussion_r456721650; we separated the test of constructAndExecuteYoutubeRequest and doGet()

Reply:
Makes sense, thanks!